src/Bicep.Core.IntegrationTests/SafeDereferenceTests.cs

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.Diagnostics.CodeAnalysis;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.UnitTests;
+using Bicep.Core.UnitTests.Assertions;
+using Bicep.Core.UnitTests.Utils;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Bicep.Core.IntegrationTests;
+
+[TestClass]
+public class SafeDereferenceTests
```

Comment:
If possible, I'd recommend uploading the latest Deployment engine to Nuget.org, picking it up in this build, and adding a test to `EvaluationTests.cs`. That'll help give a lot of confidence that the full end-to-end works (and doesn't get broken in future); expression output can be hard to review accurately.

For example, [here](https://github.com/Azure/bicep/blob/ca612fc62a66dc860644a95db5e8d9bcf2a1953a/src/Bicep.Core.IntegrationTests/EvaluationTests.cs#L38) we're able to assert that 1 + 3 actually evaluates to 4 (not just `add(2, 2)`)